v2/cli/init_command.go

```diff
@@ -310,3 +316,17 @@ func checkFileOverwrite(
 	}
 	return true, false, nil
 }
+
+func validateGitCloneURL(url string) error {
+	gitCloneURLRegex :=
+		regexp.MustCompile(
+			`((git|ssh|http(s)?)|(git@[\w\.]+))` +
+				`(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?`)
```

Comment:
Don't break the regex across two lines. It makes it harder to read. Use a `// nolint: lll` to ignore the error from the `lll` linter if you have to.

Reply:
Also, you can use https://regoio.herokuapp.com/ to test your expression.

Reply:
This regex might be more concise, readable, and performant:

```
`^(?:(?:https?://)|(?:git@))[\w:/\-\.\?=@&]*\.git$`
```

Not the use of `?:` following opening parenthesis. In regular expressions, parenthesis are used not only for grouping things, but also defining "capture groups" which are useful when you want to extract a segment of what was matched. We don't care about doing so here, so we can use `?:` to _stop_ the creation of a capture group. It will make the regex perform better.

Reply:
Further refining this, this regex is slightly better even:

```
`^(?:(?:https?://)|(?:git@))[\w:/\-\.\?=@&]+\.git$`
```

Reply:
This regex doesn't allow for an empty string for when a user doesn't use the `--git` flag. Should I just add a conditional to check for an empty string instead?

Reply:
Ya. This deliberately doesn't validate the empty string. Only check for validity if the value isn't the empty string.